### PR TITLE
SEARCH-830, fix error handling to display error message from solr

### PR DESCRIPTION
**Pull request by michaelsuzukisagi**

### CODE REVIEW

File: src/main/java/org/alfresco/repo/search/impl/solr/SolrSQLJSONResultSet.java [hunk 1]
```diff
@@ -83,7 +83,14 @@ public SolrSQLJSONResultSet(JSONObject json, BasicSearchParameters searchParamet
                 {
                     throw new RuntimeException("Unable to execute the query, this API requires InsightEngine.");
                 }
-                throw new RuntimeException("Unable to execute the query, error caused by: " + error);
+                if(searchParameters.getExtraParameters().containsKey("format"))
+                {
+
```
Comment (tunaaksoy) [inline src/main/java/org/alfresco/repo/search/impl/solr/SolrSQLJSONResultSet.java]: I think you should either add a comment here or check the negative case and leave the else block.

File: src/main/java/org/alfresco/repo/search/impl/solr/SolrQueryClient.java [hunk 2]
```diff
@@ -38,7 +38,7 @@
  */
 public interface SolrQueryClient extends BeanFactoryAware, InitializingBean
 {
-    public ResultSet executeQuery(final BasicSearchParameters searchParameters, String language);
+    public ResultSet executeQuery(final SearchParameters searchParameters, String language);
```
Comment (tunaaksoy) [inline src/main/java/org/alfresco/repo/search/impl/solr/SolrQueryClient.java]: Will this cause backwards compatibility issues?

Comment (michaelsuzukisagi) [reply src/main/java/org/alfresco/repo/search/impl/solr/SolrQueryClient.java]: No as this is a new interface and SearchParameter precedes BasicSearchParameters

File: src/main/java/org/alfresco/repo/search/impl/solr/SolrSQLJSONResultSet.java (hunk 1 above)
Comment (michaelsuzukisagi) [reply src/main/java/org/alfresco/repo/search/impl/solr/SolrSQLJSONResultSet.java]: Removed redundant if statement

**Pull request merged**